src/enclave/enclave_init.c

```diff
@@ -39,24 +39,40 @@ static void __sgxlkl_enclave_copy_app_config(
     sgxlkl_app_config_t* app_config,
     const sgxlkl_config_t* sgxlkl_config)
 {
-    int i = 0, j = 0;
+    int i = 0, envc = 0;
     char** envp = NULL;
+    size_t total_size = 0;
 
     app_config->argc = sgxlkl_config->argc;
-    app_config->argv = malloc((app_config->argc + 1) * sizeof(char*));
 
     for (i = 0; i < app_config->argc; i++)
-        app_config->argv[i] = strdup(sgxlkl_config->argv[i]);
-    app_config->argv[i] = NULL;
+        total_size += strlen(sgxlkl_config->argv[i]) + 1;
```

Comment:
Missing a check that `sgxlkl_config->argv[i]` and `sgxlkl_config->argv[i] + strlen(`sgxlkl_config->argv[i])` are both outside the enclave (and not spanning the enclave).

Reply:
See above, all those string are in enclave memory.